cirq/contrib/tpu/circuit_to_tensorflow.py

```diff
@@ -0,0 +1,463 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple, Dict, Any, List, Union, Callable, TypeVar, NamedTuple
+
+from collections import namedtuple
+
+import numpy as np
+import tensorflow as tf
+
+from cirq import ops, circuits, linalg, protocols, decompositions
+
+_GROUP_SIZE = 7
+
+
+ComputeFuncAndFeedDict = NamedTuple(
+    'ComputeFuncAndFeedDict',
+    [
+        ('compute_func', Callable[[], tf.Tensor]),
+        ('feed_dict', Dict[tf.Tensor, Any])
+    ])
+
+
+def circuit_to_tensorflow_compute_func_and_feed_dict(
```

Comment:
Yes, but also self-descriptive. I really don't want people thinking they're getting back a `tf.Tensor`.

Do you have a better name for it?

Reply:
I think your return time is pretty descriptive, and it's well documented.  I think circuit_to_tensorflow_runnable is fine (Not all graphs require feed dicts, but most of the time to run a graph in a session you do need a feed dict so it would not be unexpected to have this as part of the return type)

Reply:
Done.